bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/internal/BlueZBridgeHandler.java

```diff
@@ -0,0 +1,234 @@
+import org.openhab.binding.bluetooth.AbstractBluetoothBridgeHandler;
+import org.openhab.binding.bluetooth.BluetoothAddress;
+import org.openhab.binding.bluetooth.bluez.internal.events.AdapterDiscoveringChangedEvent;
+import org.openhab.binding.bluetooth.bluez.internal.events.AdapterPoweredChangedEvent;
+import org.openhab.binding.bluetooth.bluez.internal.events.BlueZEvent;
+import org.openhab.binding.bluetooth.bluez.internal.events.BlueZEventListener;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.hypfvieh.bluetooth.wrapper.BluetoothAdapter;
+import com.github.hypfvieh.bluetooth.wrapper.BluetoothDevice;
+
+/**
+ * The {@link BlueZBridgeHandler} is responsible for talking to the BlueZ stack, using DBus Unix Socket.
+ * This Binding does not use any JNI.
+ * It provides a private interface for {@link BlueZBluetoothDevice}s to access the stack and provides top
+ * level adaptor functionality for scanning and arbitration.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ * @author Hilbrand Bouwkamp - Simplified calling scan and better handling manual scanning
+ * @author Connor Petty - Simplified device scan logic
+ * @author Benjamin Lafois - Replaced tinyB with bluezDbus
+ *
+ */
+@NonNullByDefault
+public class BlueZBridgeHandler extends AbstractBluetoothBridgeHandler<BlueZBluetoothDevice>
+        implements BlueZEventListener {
+
+    private final Logger logger = LoggerFactory.getLogger(BlueZBridgeHandler.class);
+
+    // ADAPTER from BlueZ-DBus Library
+    private @Nullable BluetoothAdapter adapter;
+
+    // Our BT address
+    private @Nullable BluetoothAddress adapterAddress;
+
+    private @Nullable ScheduledFuture<?> discoveryJob;
+
+    private final DeviceManagerFactory deviceManagerFactory;
+
+    /**
+     * Constructor
+     *
+     * @param bridge the bridge definition for this handler
+     */
+    public BlueZBridgeHandler(Bridge bridge, DeviceManagerFactory deviceManagerFactory) {
+        super(bridge);
+        this.deviceManagerFactory = deviceManagerFactory;
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+
+        // Load configuration
+        final BlueZAdapterConfiguration configuration = getConfigAs(BlueZAdapterConfiguration.class);
+        String addr = configuration.address;
+        if (addr != null) {
+            this.adapterAddress = new BluetoothAddress(addr.toUpperCase());
+        } else {
+            // If configuration does not contain adapter address to use, exit with error.
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "address not set");
+            return;
+        }
+
+        logger.debug("Creating BlueZ adapter with address '{}'", adapterAddress);
+        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_PENDING, "Initializing");
```

Comment:
It seems you had missed this change - CONFIGURATION_PENDING is not a valid status detail here.
```suggestion
        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "Initializing");
```

Reply:
done